Change table names for better understanding.

jonimarqu opened the pull request:
Change table names for better understanding.
+ Producto into Recompensa.
+ Voluntariado into Vacante.
+ CestaRecompensas into Carrito.

+ Use Usuario.puntosAcumulados instead of Usuario.recompensasAcumuladas.
+ Add Usuario.puntosCanjeados.

jonimarqu committed:
Change table names for better understanding.
+ Producto into Recompensa.
+ Voluntariado into Vacante.
+ CestaRecompensas into Carrito.

+ Use Usuario.puntosAcumulados instead of recompensasAcumuladas.
+ Add Usuario.puntosCanjeados.

MauriDemasi commented on back/models/Usuario.js:
En este caso deberia ser puntosACanjear o algo similar, ya que la columna hace referencia al disponible de puntos que tiene el voluntario para cambiar por algun producto

jonimarqu replied:
Claro, son los puntos acumulados de cada Usuario. Me parece bien que quede Acumulados, como estaba, es bien descriptivo. Otra opción puede ser Disponibles.
Qué piensan?

jonimarqu committed:
Fix soft deletion.
- Set Carrito.name not unique.
- Set Usuario.email not unique.
- Set a beforeSave hook on a Usuario.uniqueEmail colummn to enforce uniqueness.
The hook sets uniqueEmail value to null or Usuario.email depending on deletedAt.
- Make Usuario provider not look for deleted instances.

jonimarqu committed:
Uncomnent constraint in Usuario.uniqueEmail

jonimarqu committed:
Add req.params.role to assign at User creation.

jonimarqu committed:
Move DB initialization code.
-From DB config into model index for better concern separation.

jonimarqu committed:
Fix softDeletion issue with User and Carts.
-Fix User softDeletion-hook not working issue.
-Fix User and Cart destroy methods.

jonimarqu committed:
Merge pull request #52 from jonimarqu/fix/softDeletion

Fix/soft deletion
